src/worker.cc

```diff
@@ -802,6 +805,36 @@ void from_json(rapidjson::Document& doc, Options& options) {
     options.set_resample_distance(*resample_distance);
   }
 
+  // expansion action, default to isochrone
+  auto exp_action_str = rapidjson::get_optional<std::string>(doc, "/action");
+  Options::Action exp_action;
+  if (exp_action_str) {
+    if (!Options_ExpansionAction_Enum_Parse(*exp_action_str, &exp_action)) {
+      throw valhalla_exception_t(144, *exp_action_str);
+    }
+    options.set_expansion_action(exp_action);
+  } else if (options.action() == Options_Action_expansion) {
+    throw valhalla_exception_t(115, std::string("action"));
+  }
+
+  // expansion response properties
+  auto exp_props_req = rapidjson::get_child_optional(doc, "/expansion_props");
+  auto* exp_props_pbf = options.mutable_expansion_props();
+  Options::ExpansionProps exp_prop;
+  if (exp_props_req && exp_props_req->IsArray()) {
+    for (const auto& prop : exp_props_req->GetArray()) {
+      if (!valhalla::Options_ExpansionProps_Enum_Parse(std::string(prop.GetString()), &exp_prop)) {
+        throw valhalla_exception_t(168, std::string(prop.GetString()));
+      }
+      exp_props_pbf->Add(exp_prop);
+    }
+  }
+
+  // should the expansion track opposites?
+  auto exp_skip_opps = rapidjson::get_optional<bool>(doc, "/skip_opposites");
```

Comment:
i kind of forgot about this. can you explain what this does again? it somehow doesnt track opposing edges?

Reply:
right: the problem here was that dijkstra will traverse and settle _lots_ of opposing edges which have (of course) higher cost than their previously settled counterparts. in the context of data analysis that's pretty undesirable, hence this option. you could weed out the opposites client-side, but only at super high costs.